pilot/cmd/pilot-agent/main.go

```diff
@@ -284,6 +287,8 @@ func initProxy(args []string) (*model.Proxy, error) {
 		proxy.IPAddresses = append(proxy.IPAddresses, localHostIPv4, localHostIPv6)
 	}
 
+	proxy.DiscoverIPVersions()
```

Comment:
I donot read about this, is it safe to move this?

Reply:
Yes. It is safe. It should have been here actually not later so that whenever proxy is initiated, we have ipversions discovered

Reply:
@ramaraochavali  one question for this change,  how about the situation if both proxy.SupportsIPv4() and proxy.SupportsIPv6() are true, such as in dual-stack k8s cluster?  It seems that proxy.SupportsIPv4() has the higher priority?

Reply:
Yes